course6/toyapp_thomasdhuys/drawlib/scene.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <string>
+
+#include "primitives.h"
+#include "shape.h"
+
+namespace Draw {
+
+using SceneShapes = std::vector<Shape>;
+
+class Scene {
+public:
+	SceneShapes& GetShapes();
+	const SceneShapes& GetShapes() const;
+	void Add(const Shape& shape);
+	void Add(Shape&& shape);
+private:
+	SceneShapes shapes;
+	int orientation;
+
+	friend std::ostream& operator<<(std::ostream& s , const Scene& scene);
+};
+
+std::ostream& operator<<(std::ostream& s , const Scene& ro);
+
+using Scenes = std::vector<Scene>;
+
+class SceneFactory {
+public:
+	static Scenes GetScenes();
```

Comment:
Please, remove all of this.
I this I see I know what you are trying to achieve, but the way you are going about it is just wrong.
We'll arrive them by the end of the course. For now, for yours and my sanity, just remove all of it.